packages/split-button/src/SplitButton/SplitButton.types.ts

```diff
@@ -0,0 +1,111 @@
+import { ComponentType, ReactFragment } from 'react';
+
+import {
+  type ButtonProps as ImportedButtonProps,
+  Variant as ButtonVariants,
+} from '@leafygreen-ui/button';
+import { DarkModeProps } from '@leafygreen-ui/lib';
+import { MenuItemProps } from '@leafygreen-ui/menu';
+import { type MenuProps as ImportedMenuProps } from '@leafygreen-ui/menu';
+import {
+  Align as ImportedAlign,
+  Justify as ImportedJustify,
+} from '@leafygreen-ui/popover';
+
+export const Variant = {
+  Default: ButtonVariants.Default,
+  Primary: ButtonVariants.Primary,
+  Danger: ButtonVariants.Danger,
+} as const;
+
+export type Variant = typeof Variant[keyof typeof Variant];
+
+type ButtonProps = Omit<
+  ImportedButtonProps,
+  'rightGlyph' | 'href' | 'as' | 'variant'
+>;
+
+export type MenuItemsType = ReactFragment & {
+  props: {
+    children: Array<ComponentType<MenuItemProps>>;
+  };
+};
+
+export const Align = {
```

Comment:
why are we redefining align and justify?

Reply:
I was initially doing this `export type Align = Extract<ImportedMenuProps['align'], 'top' | 'bottom'>` but that only gives me a literal union. I couldn't really do Align.Bottom., etc. Same with Justify and Variant. Is there a better way to do this?

Reply:
ah got it